[BUG] StringType can not transform to DateType when read hive table to doris by spark load

**Describe the bug**
I test doris global dictionary function by spark load.  The partition field in hive table is String type and the corresponding field in doris table is DATE type. Then I submitted a spark load job, but the job failed and I got the error message.

```
spark etl job run failed
org.apache.spark.SparkException: Job aborted due to stage failure: Task 15 in stage 13.0 failed 4 times, most recent failure: Lost task 15.3 in stage 13.0 (TID 1370, plat-data-hadoop-pro334, executor 21): java.lang.RuntimeException: Error while encoding: java.lang.RuntimeException: java.lang.String is not a valid external type for schema of date
staticinvoke(class org.apache.spark.sql.catalyst.util.DateTimeUtils$, DateType, fromJavaDate, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 0, logdate), DateType), true, false) AS logdate#240
validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 1, app_id), IntegerType) AS app_id#241
staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 2, game), StringType), true, false) AS game#242
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 3, event_action), StringType), true, false) AS event_action#243
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 4, event_category), StringType), true, false) AS event_category#244
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 5, event_label), StringType), true, false) AS event_label#245
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 6, event_value), StringType), true, false) AS event_value#246
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 7, game_level), IntegerType) AS game_level#247
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 8, game_vip_level), IntegerType) AS game_vip_level#248
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 9, game_tag), IntegerType) AS game_tag#249
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 10, event_ts), LongType) AS event_ts#250L
staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 11, uuid), StringType), true, false) AS uuid#251
	at org.apache.spark.sql.catalyst.encoders.ExpressionEncoder.toRow(ExpressionEncoder.scala:292)
	at org.apache.spark.sql.SparkSession$$anonfun$4.apply(SparkSession.scala:593)
	at org.apache.spark.sql.SparkSession$$anonfun$4.apply(SparkSession.scala:593)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:435)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:441)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at org.apache.spark.util.collection.ExternalSorter.insertAll(ExternalSorter.scala:191)
	at org.apache.spark.shuffle.sort.SortShuffleWriter.write(SortShuffleWriter.scala:62)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: java.lang.String is not a valid external type for schema of date
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.StaticInvoke_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.writeFields_0_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.apply(Unknown Source)
	at org.apache.spark.sql.catalyst.encoders.ExpressionEncoder.toRow(ExpressionEncoder.scala:289)
	... 20 more

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1889)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1877)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1876)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1876)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:926)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2110)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2059)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2048)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:737)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:935)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:933)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)
	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:933)
	at org.apache.spark.api.java.JavaRDDLike$class.foreachPartition(JavaRDDLike.scala:219)
	at org.apache.spark.api.java.AbstractJavaRDDLike.foreachPartition(JavaRDDLike.scala:45)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.writeRepartitionAndSortedRDDToParquet(SparkDpp.java:194)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.processRollupTree(SparkDpp.java:332)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.process(SparkDpp.java:1014)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.doDpp(SparkDpp.java:1050)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.processDpp(SparkEtlJob.java:142)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.processData(SparkEtlJob.java:226)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.run(SparkEtlJob.java:233)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.main(SparkEtlJob.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.yarn.ApplicationMaster$$anon$2.run(ApplicationMaster.scala:684)
Caused by: java.lang.RuntimeException: Error while encoding: java.lang.RuntimeException: java.lang.String is not a valid external type for schema of date

```


**Expected behavior**
The String type field can be transform to DATE type  automatically.

**Screenshots**
In the function convertSrcDataframeToDstDataframe(xxx), it seem to do transform string to date, but it can not work.

![image](https://user-images.githubusercontent.com/26566636/104296590-c6e01c80-54fc-11eb-8d5a-92f69f6a7b5a.png)
![image](https://user-images.githubusercontent.com/26566636/104296357-7bc60980-54fc-11eb-94a4-d4a2be033ea7.png)
![image](https://user-images.githubusercontent.com/26566636/104296386-81235400-54fc-11eb-8cd8-bf97d832444c.png)

**Desktop (please complete the following information):**
 - OS: [centos7]
 - Version [0.13.0-rc03 ]

--- a/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java
+++ b/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java
@@ -34,8 +34,8 @@ public abstract class ColumnParser implements Serializable {
     protected static final Logger LOG = LogManager.getLogger(ColumnParser.class);
 
     // thread safe formatter
-    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("YYYY-MM-DD");
-    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.forPattern("YYYY-MM-DD HH:MM:SS");
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd");
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
 
     public static ColumnParser create(EtlJobConfig.EtlColumn etlColumn) throws SparkDppException {
         String columnType = etlColumn.columnType;

--- a/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java
+++ b/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java
@@ -21,17 +21,21 @@ import org.apache.doris.common.SparkDppException;
 import org.apache.doris.load.loadv2.etl.EtlJobConfig;
 import org.apache.log4j.LogManager;
 import org.apache.log4j.Logger;
-import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
 
 import java.io.Serializable;
 import java.math.BigDecimal;
 import java.math.BigInteger;
-import java.util.Date;
 
 // Parser to validate value for different type
 public abstract class ColumnParser implements Serializable {
 
     protected static final Logger LOG = LogManager.getLogger(ColumnParser.class);
+
+    // thread safe formatter
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("YYYY-MM-DD");
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.forPattern("YYYY-MM-DD HH:MM:SS");
 
     public static ColumnParser create(EtlJobConfig.EtlColumn etlColumn) throws SparkDppException {
         String columnType = etlColumn.columnType;
@@ -158,7 +162,7 @@ class DateParser extends ColumnParser {
     @Override
     public boolean parse(String value) {
         try {
-            Date.parse(value);
+            DATE_FORMATTER.parseDateTime(value);
         } catch (IllegalArgumentException e) {
             return false;
         }
@@ -170,7 +174,7 @@ class DatetimeParser extends ColumnParser {
     @Override
     public boolean parse(String value) {
         try {
-            DateTime.parse(value);
+            DATE_TIME_FORMATTER.parseDateTime(value);
         } catch (IllegalArgumentException e) {
             return false;
         }

--- a/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java
+++ b/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java
@@ -890,7 +890,7 @@ public final class SparkDpp implements java.io.Serializable {
             // case 2: bitmap hash function; this func is not supported in spark load now, so ignore it here
             // case 3: origin value is a integer value; it should be checked use LongParser
             if (StringUtils.equalsIgnoreCase(column.columnType, "bitmap")) {
-                if (dictBitmapColumnSet.contains(column.columnName)) {
+                if (dictBitmapColumnSet.contains(column.columnName.toLowerCase())) {
                     continue;
                 } else {
                     columnNameNeedCheckArrayList.add(column);

--- a/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java
+++ b/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java
@@ -108,13 +108,17 @@ public final class SparkDpp implements java.io.Serializable {
     // we need to wrap it so that we can use it in executor.
     private SerializableConfiguration serializableHadoopConf;
     private DppResult dppResult = new DppResult();
+    Map<Long, Set<String>> tableToBitmapDictColumns = new HashMap<>();
 
     // just for ut
     public SparkDpp() {}
 
-    public SparkDpp(SparkSession spark, EtlJobConfig etlJobConfig) {
+    public SparkDpp(SparkSession spark, EtlJobConfig etlJobConfig, Map<Long, Set<String>> tableToBitmapDictColumns) {
         this.spark = spark;
         this.etlJobConfig = etlJobConfig;
+        if (tableToBitmapDictColumns != null) {
+            this.tableToBitmapDictColumns = tableToBitmapDictColumns;
+        }
     }
 
     public void init() {
@@ -543,7 +547,9 @@ public final class SparkDpp implements java.io.Serializable {
                 }
             }
             if (column.columnType.equalsIgnoreCase("DATE")) {
-                dataframe = dataframe.withColumn(dstField.name(), dataframe.col(dstField.name()).cast("date"));
+                dataframe = dataframe.withColumn(dstField.name(), dataframe.col(dstField.name()).cast(DataTypes.DateType));
+            } else if (column.columnType.equalsIgnoreCase("DATETIME")) {
+                dataframe = dataframe.withColumn(dstField.name(), dataframe.col(dstField.name()).cast(DataTypes.TimestampType));
             } else if (column.columnType.equalsIgnoreCase("BOOLEAN")) {
                 dataframe = dataframe.withColumn(dstField.name(),
                         functions.when(functions.lower(dataframe.col(dstField.name())).equalTo("true"), "1")
@@ -844,7 +850,8 @@ public final class SparkDpp implements java.io.Serializable {
                                                String hiveDbTableName,
                                                EtlJobConfig.EtlIndex baseIndex,
                                                EtlJobConfig.EtlFileGroup fileGroup,
-                                               StructType dstTableSchema) throws SparkDppException {
+                                               StructType dstTableSchema,
+                                               Set<String> dictBitmapColumnSet) throws SparkDppException {
         // select base index columns from hive table
         StringBuilder sql = new StringBuilder();
         sql.append("select ");
@@ -857,18 +864,39 @@ public final class SparkDpp implements java.io.Serializable {
         }
 
         Dataset<Row> dataframe = spark.sql(sql.toString());
+        // Note(wb): in current spark load implementation, spark load can't be consistent with doris BE; The reason is as follows
+        // For stream load in doris BE, it runs as follow steps:
+        // step 1: type check
+        // step 2: expression calculation
+        // step 3: strict mode check
+        // step 4: nullable column check
+        // BE can do the four steps row by row
+        // but spark load relies on spark to do step2, so it can only do step 1 for whole dataset and then do step 2 for whole dataset and so on;
+        // So in spark load, we first do step 1,3,4,and then do step 2.
         dataframe = checkDataFromHiveWithStrictMode(dataframe, baseIndex, fileGroup.columnMappings.keySet(), etlJobConfig.properties.strictMode,
-                    dstTableSchema);
+                dstTableSchema, dictBitmapColumnSet);
         dataframe = convertSrcDataframeToDstDataframe(baseIndex, dataframe, dstTableSchema, fileGroup);
         return dataframe;
     }
 
     private Dataset<Row> checkDataFromHiveWithStrictMode(
-            Dataset<Row> dataframe, EtlJobConfig.EtlIndex baseIndex, Set<String> mappingColKeys, boolean isStrictMode, StructType dstTableSchema) throws SparkDppException {
+            Dataset<Row> dataframe, EtlJobConfig.EtlIndex baseIndex, Set<String> mappingColKeys, boolean isStrictMode, StructType dstTableSchema,
+            Set<String> dictBitmapColumnSet) throws SparkDppException {
         List<EtlJobConfig.EtlColumn> columnNameNeedCheckArrayList = new ArrayList<>();
         List<ColumnParser> columnParserArrayList = new ArrayList<>();
         for (EtlJobConfig.EtlColumn column : baseIndex.columns) {
-            if (!StringUtils.equalsIgnoreCase(column.columnType, "varchar") &&
+            // note(wb): there are three data source for bitmap column
+            // case 1: global dict; need't check
+            // case 2: bitmap hash function; this func is not supported in spark load now, so ignore it here
+            // case 3: origin value is a integer value; it should be checked use LongParser
+            if (StringUtils.equalsIgnoreCase(column.columnType, "bitmap")) {
+                if (dictBitmapColumnSet.contains(column.columnName)) {
+                    continue;
+                } else {
+                    columnNameNeedCheckArrayList.add(column);
+                    columnParserArrayList.add(new BigIntParser());
+                }
+            } else if (!StringUtils.equalsIgnoreCase(column.columnType, "varchar") &&
                     !StringUtils.equalsIgnoreCase(column.columnType, "char") &&
                     !mappingColKeys.contains(column.columnName)) {
                 columnNameNeedCheckArrayList.add(column);
@@ -879,6 +907,7 @@ public final class SparkDpp implements java.io.Serializable {
         ColumnParser[] columnParserArray = columnParserArrayList.toArray(new ColumnParser[columnParserArrayList.size()]);
         EtlJobConfig.EtlColumn[] columnNameArray = columnNameNeedCheckArrayList.toArray(new EtlJobConfig.EtlColumn[columnNameNeedCheckArrayList.size()]);
 
+        StructType srcSchema = dataframe.schema();
         JavaRDD<Row> result = dataframe.toJavaRDD().flatMap(new FlatMapFunction<Row, Row>() {
             @Override
             public Iterator<Row> call(Row row) throws Exception {
@@ -898,6 +927,11 @@ public final class SparkDpp implements java.io.Serializable {
                         if (isStrictMode) {
                             validRow = false;
                             LOG.warn(String.format("row parsed failed in strict mode, column name %s, src row %s", column.columnName, row.toString()));
+                        // a column parsed failed would be filled null, but if doris column is not allowed null, we should skip this row
+                        } else if (!column.isAllowNull) {
+                            validRow = false;
+                            LOG.warn("column:" + i + " can not be null. row:" + row.toString());
+                            break;
                         } else {
                             columnIndexNeedToRepalceNull.add(fieldIndex);
                         }
@@ -909,7 +943,7 @@ public final class SparkDpp implements java.io.Serializable {
                     if (abnormalRowAcc.value() <= 5) {
                         invalidRows.add(row.toString());
                     }
-                } if (columnIndexNeedToRepalceNull.size() != 0) {
+                } else if (columnIndexNeedToRepalceNull.size() != 0) {
                     Object[] newRow = new Object[row.size()];
                     for (int i = 0; i < row.size(); i++) {
                         if (columnIndexNeedToRepalceNull.contains(i)) {
@@ -926,7 +960,8 @@ public final class SparkDpp implements java.io.Serializable {
             }
         });
 
-        return spark.createDataFrame(result, dstTableSchema);
+        // here we just check data but not do cast, so data type should be same with src schema which is hive table schema
+        return spark.createDataFrame(result, srcSchema);
     }
 
     private void process() throws Exception {
@@ -934,6 +969,7 @@ public final class SparkDpp implements java.io.Serializable {
             for (Map.Entry<Long, EtlJobConfig.EtlTable> entry : etlJobConfig.tables.entrySet()) {
                 Long tableId = entry.getKey();
                 EtlJobConfig.EtlTable etlTable = entry.getValue();
+                Set<String> dictBitmapColumnSet = tableToBitmapDictColumns.getOrDefault(tableId, new HashSet<>());
 
                 // get the base index meta
                 EtlJobConfig.EtlIndex baseIndex = null;
@@ -982,7 +1018,8 @@ public final class SparkDpp implements java.io.Serializable {
                     if (sourceType == EtlJobConfig.SourceType.FILE) {
                         fileGroupDataframe = loadDataFromFilePaths(spark, baseIndex, filePaths, fileGroup, dstTableSchema);
                     } else if (sourceType == EtlJobConfig.SourceType.HIVE) {
-                        fileGroupDataframe = loadDataFromHiveTable(spark, fileGroup.dppHiveDbTableName, baseIndex, fileGroup, dstTableSchema);
+                        fileGroupDataframe = loadDataFromHiveTable(spark, fileGroup.dppHiveDbTableName, baseIndex, fileGroup, dstTableSchema,
+                                dictBitmapColumnSet);
                     } else {
                         throw new RuntimeException("Unknown source type: " + sourceType.name());
                     }

--- a/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/etl/SparkEtlJob.java
+++ b/fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/etl/SparkEtlJob.java
@@ -17,6 +17,7 @@
 
 package org.apache.doris.load.loadv2.etl;
 
+import org.apache.doris.common.SparkDppException;
 import org.apache.doris.load.loadv2.dpp.GlobalDictBuilder;
 import org.apache.doris.load.loadv2.dpp.SparkDpp;
 import org.apache.doris.load.loadv2.etl.EtlJobConfig.EtlColumn;
@@ -53,6 +54,7 @@ public class SparkEtlJob {
 
     private static final String BITMAP_DICT_FUNC = "bitmap_dict";
     private static final String TO_BITMAP_FUNC = "to_bitmap";
+    private static final String BITMAP_HASH = "bitmap_hash";
 
     private String jobConfigFilePath;
     private EtlJobConfig etlJobConfig;
@@ -112,8 +114,11 @@ public class SparkEtlJob {
                     String columnName = mappingEntry.getKey();
                     String exprStr = mappingEntry.getValue().toDescription();
                     String funcName = functions.expr(exprStr).expr().prettyName();
+                    if (funcName.equalsIgnoreCase(BITMAP_HASH)) {
+                        throw new SparkDppException("spark load not support bitmap_hash now");
+                    }
                     if (funcName.equalsIgnoreCase(BITMAP_DICT_FUNC)) {
-                        bitmapDictColumns.add(columnName);
+                        bitmapDictColumns.add(columnName.toLowerCase());
                     } else if (!funcName.equalsIgnoreCase(TO_BITMAP_FUNC)) {
                         newColumnMappings.put(mappingEntry.getKey(), mappingEntry.getValue());
                     }
@@ -137,7 +142,7 @@ public class SparkEtlJob {
     }
 
     private void processDpp() throws Exception {
-        SparkDpp sparkDpp = new SparkDpp(spark, etlJobConfig);
+        SparkDpp sparkDpp = new SparkDpp(spark, etlJobConfig, tableToBitmapDictColumns);
         sparkDpp.init();
         sparkDpp.doDpp();
     }

